buildtest/executors/base.py

```diff
@@ -0,0 +1,102 @@
+"""
+BuildExecutor: manager for test executors
+Copyright (C) 2020 Vanessa Sochat. 
+"""
+
+import os
+import sys
+
+from buildtest.config import config_opts
+from buildtest.log import init_logfile, init_log
+
+
+class BuildExecutor:
+    """A BuildExector is a base class some type of executor, defined under
+       the buildtest/settings/default-config.json schema. For example,
+       the types "local" and "slurm" would map to `LocalExecutor` and
+       `SlurmExecutor` here, each expecting a particular set of
+       variables under the config options. If options are required
+       and not provided, we exit on error. If they are optional and not
+       provided, we use reasonable defaults.
+    """
+
+    def __init__(self, config_opts):
+        """initiate executors, meaning that we provide the config_opts
+           that are validated, and can instantiate each executor to be available
+           
+           Parameters:
+
+           config_opts: the validated config opts provided by buildtest.
```

Comment:
Can we make use of ``param`` and ``type``. See https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html for list of docstring format supported by Sphinx. You can see how i did this for lmodule at https://github.com/HPC-buildtest/lmodule/blob/master/lmod/module.py#L51 

It makes for some nice API docs

Reply:
Sure I can change that.